src/test/java/com/powsybl/openloadflow/util/LoadFlowAssert.java

```diff
@@ -47,4 +50,22 @@ public static void assertUndefinedActivePower(Terminal terminal) {
     public static void assertUndefinedReactivePower(Terminal terminal) {
         assertTrue(Double.isNaN(terminal.getQ()));
     }
+
+    public static void assertBetterResults(LoadFlowResult loadFlowResult, LoadFlowResult loadFlowResultBetter) {
+        assertTrue(loadFlowResult.isOk(), "results should be ok");
+        assertTrue(loadFlowResultBetter.isOk(), "results should be ok");
+        assertEquals(loadFlowResult.getComponentResults().size(), loadFlowResultBetter.getComponentResults().size(), "results should have same subnetwork count");
+        Iterator<LoadFlowResult.ComponentResult> componentResultIterator = loadFlowResult.getComponentResults().iterator();
+        Iterator<LoadFlowResult.ComponentResult> componentResultIteratorBetter = loadFlowResultBetter.getComponentResults().iterator();
+        // loop over sub networks
+        while (componentResultIterator.hasNext()) {
+            LoadFlowResult.ComponentResult componentResult = componentResultIterator.next();
+            LoadFlowResult.ComponentResult componentResultBetter = componentResultIteratorBetter.next();
+            assertEquals(componentResult.getComponentNum(), componentResultBetter.getComponentNum(), "this assert has a bug, please fix it");
+            assertEquals(componentResult.getSlackBusId(), componentResultBetter.getSlackBusId(), "this assert has a bug, please fix it");
+            assertEquals(componentResult.getStatus(), componentResultBetter.getStatus(), "status results should be the same");
+            assertEquals(componentResult.getIterationCount(), componentResultBetter.getIterationCount(), "iteration count results should be the same");
+            assertTrue(Math.abs(componentResult.getSlackBusActivePowerMismatch()) > Math.abs(componentResultBetter.getSlackBusActivePowerMismatch()), "it was expected that improved process returns a better result");
```

Comment:
Why are you not doing here an assertEquals ?

Reply:
It seems ending **mismatch** is smaller by making `power factor` a constant value.
Is it true ?
That's why my assert was called **assertBetterResults**.

Reply:
I'm not sure this is true for all cases, but just a lucky side effect